### DISCUSSION TITLE
Init Vehicle Failed, TERMINATE! DJI M300 OSDK

### DISCUSSION THREAD

**Discussion by DevJadefly** [community/community#120993] [Programming Help]
### Body

Hello, I am trying to figure out a way to run OSDK sample code provided by DJI on my M300 with ros noetic. I am working with OSDK and the E-port development kit by DJI so I plugged the UART and the USB ports to my computer (ubuntu 20.04). My worflow : 
I plug the aircraft to the E-port development kit, then the UART and USB ports to my computer.
I turn on the remote controller and go to the pilot app 
I turn on the aircraft (the aircraft is connected to the remote)
I tried to run the sample code with this line :
roslaunch dji_osdk_ros dji_vehicle_node.launch

But I can't make it work it keeps giving this error : 
[273648.667]STATUS/1 @ init, L247: USB is not plugged or initialized successfully. Advacned-Sensing will not run.
[273648.667]STATUS/1 @ activate, L1329: version 0x304122A

[273648.667]STATUS/1 @ firewallTask, L112: [273649.689]STATUS/1 @ activate, L1360: Retry to activate again ..
[273650.722]STATUS/1 @ activate, L1360: Retry to activate again ..
[273651.749]STATUS/1 @ activate, L1360: Retry to activate again ..
[273651.776]STATUS/1 @ getErrorCodeMessage, L744: activate
[273651.777]STATUS/1 @ getCMDSetActivationMSG, L794: NEW_DEVICE_ERROR

[273651.777]ERRORLOG/1 @ activate, L1380: Failed to activate please retry SET 0x0 ID 0x1 code 0x3

[273651.777]ERRORLOG/1 @ activate, L1387: Solutions for NEW_DEVICE_ERROR:
	* Double-check your app_id and app_key in UserConfig.txt. Does it match with your DJI developer account?
	* If this is a new device, you need to activate it through the App or DJI Assistant 2 with Internet
	For different aircraft, the App and the version of DJI Ass
[273651.777]STATUS/1 @ getErrorCodeMessage, L744: initVehicle
[273651.777]STATUS/1 @ getCMDSetActivationMSG, L794: NEW_DEVICE_ERROR

Init Vehicle Failed, TERMINATE!

I checked my app ID and Key, the app does not request any activation ...
Has anyone run into the same problem ? 

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by kingkhangkang** [comment]
Heya! If you ever managed to fix that/would be able to bypass the DJI ronin activation issue in the assistant App, please let me know. Would be willing to commission!

**Reply by DevJadefly** [reply]
Hello ! Thanks for the answer. Actually, the problem was that it was not enough to have the pilot app opened on the RC so I had to have DJI Assistant 2 opened as well when launching dji_vehicle_node.launch. A pop up appeared on DJIA2 to activate the aircraft. It was only needed the first time using OSDK with this aircaft.

**Comment by aradfarahani** [comment]
It sounds like you're encountering a common issue with the DJI OSDK setup. Here are some steps and checks you can follow to troubleshoot and resolve the problem:

### Troubleshooting Steps

1. **Check App ID and App Key**:
   - Ensure that your `app_id` and `app_key` in the `UserConfig.txt` file match exactly with those in your DJI developer account. Even a small typo can cause issues.

2. **Device Activation**:
   - If this is a new device, you need to activate it through the DJI Pilot App or DJI Assistant 2 with an internet connection. Make sure you are logged into the DJI account that created the app ID.

3. **USB Connection**:
   - Verify that the USB connection is secure and that the USB port is functioning correctly. Try using a different USB port or cable if necessary.

4. **Permissions**:
   - Ensure that your user has the necessary permissions to access the USB and UART ports. You can do this by running:
     ```bash
     sudo chmod a+rw /dev/ttyUSB0
     sudo usermod -a -G dialout $USER
     ```
   - After running these commands, you may need to log out and log back in for the changes to take effect.

5. **Baud Rate**:
   - Check that the baud rate in the `UserConfig.txt` file matches the baud rate set in DJI Assistant 2. For some devices, only specific baud rates work well (e.g., 1000000 for M210 V2).

6. **Serial Device**:
   - Ensure you are using the correct serial device. It is recommended to use the `/dev/serial/by-id/usb-XXX` method to avoid issues with device naming.

7. **Advanced Sensing**:
   - The error message indicates an issue with advanced sensing. Make sure the advanced sensing feature is enabled and properly configured in the DJI Pilot App or DJI Assistant 2.

8. **Retry Activation**:
   - Sometimes, simply retrying the activation process can resolve the issue. Ensure that the remote controller and aircraft are both turned on and connected.

### Example Configuration

Here is an example of what your `UserConfig.txt` might look like:
```plaintext
app_id : 123456
app_key : abcdefghijklmnopqrstuvwxyz
device : /dev/ttyUSB0
baudrate : 230400
```

### Additional Resources

- **DJI Forum**: [[OSDK and OSDK-ros activation failure with solution and checklist](https://forum.dji.com/thread-216529-1-1.html)](https://forum.dji.com/thread-216529-1-1.html)[ provides a comprehensive checklist for troubleshooting common issues](https://forum.dji.com/thread-216529-1-1.html).
- **GitHub**: The [dji-sdk/Onboard-SDK-ROS](https://github.com/dji-sdk/Onboard-SDK-ROS) repository contains official ROS packages and documentation.

By following these steps, you should be able to identify and resolve the issue. If the problem persists, consider reaching out to the DJI support community or checking the DJI forums for additional insights. Good luck! If you need further assistance, feel free to ask.
